TelemetrySite/server/Context/context.py

```diff
@@ -0,0 +1,231 @@
+                            hasIMU = False
+                        else:
+                            hasTMU = False
+                        continue
+                    format_array = [Identifier(self.configNames[index].lower())]
+                    conn = utils.connect()
+                    cur = conn.cursor()
+                    config_name = [currentConfigData["selected"]]
+
+                    cur.execute(
+                        SQL("SELECT id FROM {} WHERE configName = %s").format(
+                            *format_array
+                        ),
+                        (*config_name,),
+                    )
+                    idBikeConfig.append(cur.fetchone()[0])
+
+                    conn.commit()
+                    conn.close()
+
+        # establish sql commands for all non event, bike, and context configs
+        # save the event and bike ids for to reference in the context db
+
+        sqlEvent = "INSERT into Event (id, eventDate, eventType, location) VALUES (DEFAULT, %s, %s, %s) RETURNING id"
+
+        eventAndBikeId.append(
+            utils.exec_commit_with_id(
+                sqlEvent, self.DictToTuple(contextValue["Event"])
+            )[0][0]
+        )
+
+        # if the bike isn't a saved value
+        if not bikeSaved:
+            # create a new one
+            sqlBikeConfig = """INSERT into BikeConfig (id,  platformName, tirePressure, 
+            coolantVolume, configName, bmsConfigId, imuConfigId, tmuConfigId, tmsConfigId, 
+            pvcConfigId, mcConfigId) VALUES (DEFAULT,  """
+
+            contextBodyValues = []
+            # loop to see if tirePressure and coolantVolume are null
+            for value in self.DictToTuple(contextValue["BikeConfig"]):
+                if not value == "":
+                    contextBodyValues.append(value)
+                    sqlBikeConfig += "%s, "
+                else:
+                    sqlBikeConfig += "DEFAULT, "
+
+            # config name and bmsConfig format
+            sqlBikeConfig += "%s, "
+
+            # set imu and tmu to default as needed
+            if hasIMU is True:
+                sqlBikeConfig += "%s, "
+            else:
+                sqlBikeConfig += "DEFAULT, "
+
+            if hasTMU is True:
+                sqlBikeConfig += "%s, "
+            else:
+                sqlBikeConfig += "DEFAULT, "
+            # add tms, pvc, and mc
+            sqlBikeConfig += "%s, %s, %s) RETURNING id"
+            print(sqlBikeConfig)
+            print(tuple(contextBodyValues) + tuple(idBikeConfig))
+            eventAndBikeId.append(
+                utils.exec_commit_with_id(
+                    sqlBikeConfig,
+                    tuple(contextBodyValues) + tuple(idBikeConfig),
+                )[0][0]
+            )
+
+        else:
+            # if it is saved, get the id
+            sqlCommand = f"SELECT id from BikeConfig WHERE configName = '{contextValue["BikeConfig"]["selected"]}'"
```

Comment:
@tmb7852 Is this f-string unavoidable?

Reply:
I just switched it. I must of missed it when I switched the others to %s form